### PR TITLE
Nks 954 istio mesh acceptance

### PR DESCRIPTION
**Pull request by aajdinov**
istio mesh test output:
```
$ GO111MODULE=on make testacc TESTARGS='-run=TestAccResourceIstioMesh_Basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccResourceIstioMesh_Basic -timeout 120m
?   	github.com/NetApp/terraform-provider-nks	[no test files]
=== RUN   TestAccResourceIstioMesh_Basic
--- PASS: TestAccResourceIstioMesh_Basic (1247.11s)
PASS
ok  	github.com/NetApp/terraform-provider-nks/nks	1247.118s

```

datasource_workspace test output:
```
$ GO111MODULE=on make testacc TESTARGS='-run=TestAccDataSourceWorkspace_lookup'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccDataSourceWorkspace_lookup -timeout 120m
?   	github.com/NetApp/terraform-provider-nks	[no test files]
=== RUN   TestAccDataSourceWorkspace_lookup
--- PASS: TestAccDataSourceWorkspace_lookup (7.59s)
PASS
ok  	github.com/NetApp/terraform-provider-nks/nks	7.602s
```

### CODE REVIEW

File: nks/resource_nks_istio_mesh_test.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	org_id     = "${data.nks_organization.org.id}"
+	cluster_id = "${nks_cluster.terraform-cluster-b.id}"
+	solution   = "istio"
+}
+
+resource "nks_istio_mesh" "istio-mesh-tf" {
+	name        = "tf-istio-mesh-1"
+	mesh_type   = "cross_cluster"
+	org_id      = "${data.nks_organization.org.id}"
+	workspace   = "105"
```
Comment (jasmingacic) [inline nks/resource_nks_istio_mesh_test.go]: Pull default workspace ID using workspace datasource.

File: nks/resource_nks_istio_mesh.go [hunk 2]
```diff
@@ -78,19 +83,39 @@ func resourceNKSIstioMeshCreate(d *schema.ResourceData, meta interface{}) error
...
 	}
 
+	timeout := int(d.Timeout("Create").Seconds())
+	if v, ok := d.GetOk("timeout"); ok {
+		timeout = v.(int)
+	}
+
+	for i, solutionID := range solutionIds {
+		if err := config.Client.WaitSolutionInstalled(orgID, im.Members[i].Cluster, solutionID, timeout); err != nil {
+			log.Printf("[DEBUG] Solution %d create failed while waiting for 'installed' state: %s\n", (solutionID), err)
```
Comment (jasmingacic) [inline nks/resource_nks_istio_mesh.go]: put this into fmt.Errorf and return it at the end instead of err

**Comment by jasmingacic** [comment]
This is looking good. Once you have addressed the requested changes I will get it merged.

File: nks/resource_nks_istio_mesh_test.go (hunk 1 above)
Comment (aajdinov) [reply nks/resource_nks_istio_mesh_test.go]: Thank you for the review and the feedback. The requested change was implemented.

File: nks/resource_nks_istio_mesh.go (hunk 2 above)
Comment (aajdinov) [reply nks/resource_nks_istio_mesh.go]: Thanks for the feedback. Done.

**Pull request merged**